src/main/java/org/jenkinsci/plugins/deployment/DeploymentTrigger.java

```diff
@@ -101,10 +107,15 @@ private Triggered findTriggeredRecord(Fingerprint f) {
         public void onChange(final DeploymentFacet facet, HostRecord newRecord) {
             POOL.submit(new Runnable() {
                 public void run() {
-                    for (AbstractProject<?,?> p : Jenkins.getInstance().getAllItems(AbstractProject.class)) {
-                        DeploymentTrigger t = p.getTrigger(DeploymentTrigger.class);
-                        if (t!=null) {
-                            t.checkAndFire(facet);
+                    for (Job<?,?> job : Jenkins.getInstance().getAllItems(Job.class)) {
```

Comment:
Add a note to use the new idiom in 1.621: `TODO - 1.621: use getTrigger(Job<?,?> job, Class<T> clazz)`